contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,36 +130,220 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
-        return ();
+    let (loser_relics_len) = owned_relics_len.read(loser_token_id);
+
+    let (remove_relics_len) = loop_claim_order_relic(
+        0, realms_address, winner_realm_data, loser_token_id, loser_relics_len, 0, remove_order_relics
+    );
+
+    return loop_remove_relic(0, 0, loser_relics_len, loser_token_id, remove_relics_len, remove_order_relics);
+}
+
+// @notice loop loser relic and return if order of winner matches holder
+// @implicit syscall_ptr
+// @implicit range_check_ptr
+// @param index: index of owner relics to check, starting at 0
+// @param realms_address: realms token address
+// @param winner_realm_data: realm data of the winner
+// @param loser_token_id: realm id of loser
+// @param loser_relics_len: length of relics array of loser
+// @param remove_relics_index: index of relics to remove
+// @param remove_relics: array of relics to remove from loser
+// @return remove_relics_len: length of relics to remove from loser 
+func loop_claim_order_relic{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_check_ptr}(
+    index: felt,
+    realms_address: felt,
+    winner_realm_data: RealmData,
+    loser_token_id: Uint256,
+    loser_relics_len: felt,
+    remove_relics_index: felt,
+    remove_relics: Uint256*,
+) -> (remove_relics_len: felt) {
+    alloc_locals;
+    if (index == loser_relics_len) {
+        return (remove_relics_index,);
     }
 
-    // TODO: Add Order capture back
+    let (relic_id) = owned_relics.read(loser_token_id, index);
 
+    let (relic_owner_data: RealmData) = IRealms.fetch_realm_data(realms_address, relic_id);
+
+    if (winner_realm_data.order == relic_owner_data.order) {
+        _set_relic_holder(relic_id, relic_id);
+        assert remove_relics[remove_relics_index] = relic_id;
+        return loop_claim_order_relic(
+            index + 1,
+            realms_address,
+            winner_realm_data,
+            loser_token_id,
+            loser_relics_len,
+            remove_relics_index + 1,
+            remove_relics,
+        );
+    }
+    return loop_claim_order_relic(
+        index + 1,
+        realms_address,
+        winner_realm_data,
+        loser_token_id,
+        loser_relics_len,
+        remove_relics_index,
+        remove_relics,
+    );
+}
+
+// @notice return held relics to original owners
+// @implicit syscall_ptr
+// @implicit range_check_ptr
+// @param realms_token_id: realm id that needs assets returned
+@external
+func return_relics{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    realms_token_id: Uint256
+) {
+    alloc_locals;
+    // Only Settling
+    Module.only_approved();
+    let (relics_len) = owned_relics_len.read(realms_token_id);
```

Comment:
Shouldn't this be reset to 0 for the `realms_token_id` at some point?

Reply:
`loop_return_relics` function builds the `remove_relics` array (being all the relics they own) and the count is reset `loop_remove_relic`. 

in `loop_remove_relic`:

`owned_relics_len.write(owner_token_id, relics_len - remove_relics_len);` where `remove_relics_len` is the same as `relics_len` in this case